libcodechecker/cmd/cmd_line_client.py

```diff
@@ -259,54 +262,198 @@ def handle_list_results(args):
 
         while results:
             all_results.extend(results)
             offset += limit
-            results = getterFn(baseid, newid, limit, offset, sort_type,
+            results = getterFn(baseid, newid, limit, offset, sort_mode,
                                report_filter)
+        return all_results
+
+    def getReportDirResults(reportdir):
+        all_reports = []
+        for filename in os.listdir(reportdir):
+            if filename.endswith(".plist"):
+                file_path = os.path.join(reportdir, filename)
+                LOG.debug("Parsing:" + file_path)
+                try:
+                    files, reports = plist_parser.parse_plist(file_path)
+                    for report in reports:
+                        report.main['location']['file_name'] = \
+                            files[int(report.main['location']['file'])]
+                    all_reports.extend(reports)
+
+                except Exception as ex:
+                    LOG.error('The generated plist is not valid!')
+                    LOG.error(ex)
+        return all_reports
+
+    def getLineFromFile(filename, lineno):
+        with open(filename, 'r') as f:
+            i = 1
+            for line in f:
+                if i == lineno:
+                    return line
+                i += 1
+        return ""
+
+    def getLineFromRemoteFile(client, fid, lineno):
+        # FIXME: large special file content
+        # cause connection abort
+        source = client.getSourceFileData(fid, True)
+        i = 1
+        lines = source.fileContent.split('\n')
+        for line in lines:
+                if i == lineno:
+                    return line
+                i += 1
+        return ""
+
+    def getDiffReportDir(getterFn, baseid, report_dir, suppr, diff_type):
+        report_filter = [
+            codeCheckerDBAccess.ttypes.ReportFilter(suppressed=suppr)]
+        add_filter_conditions(report_filter[0], args.filter)
+
+        sort_mode = []
+        sort_mode.append(codeCheckerDBAccess.ttypes.SortMode(
+            codeCheckerDBAccess.ttypes.SortType.FILENAME,
+            codeCheckerDBAccess.ttypes.Order.ASC))
+        limit = codeCheckerDBAccess.constants.MAX_QUERY_SIZE
+        offset = 0
 
+        base_results = []
+        results = getterFn(baseid, limit, offset, sort_mode,
+                           report_filter)
+        while results:
+            base_results.extend(results)
+            offset += limit
+            results = getterFn(baseid, limit, offset, sort_mode,
+                               report_filter)
+        base_hashes = {}
+        for res in base_results:
+            base_hashes[res.bugHash] = res
+
+        filtered_reports = []
+        new_results = getReportDirResults(report_dir)
+        new_hashes = {}
+        for rep in new_results:
+            bughash = rep.main['issue_hash_content_of_line_in_context']
+            new_hashes[bughash] = rep
+        # show new reports from the report dir
```

Comment:
Please move the comments into the branch's body, and format them as full sentence.